rig/machine_control/bmp_controller.py

```diff
@@ -0,0 +1,422 @@
+                             response.data)
+
+        return ADCInfo(
+            v1_2c=data[1] * BMP_V_SCALE_2_5,
+            v1_2b=data[2] * BMP_V_SCALE_2_5,
+            v1_2a=data[3] * BMP_V_SCALE_2_5,
+            v1_8=data[4] * BMP_V_SCALE_2_5,
+            v3_3=data[6] * BMP_V_SCALE_3_3,
+            vpwr=data[7] * BMP_V_SCALE_12,
+            temp_top=float(data[8]) * BMP_TEMP_SCALE,
+            temp_btm=float(data[9]) * BMP_TEMP_SCALE,
+            temp_ext_0=((float(data[12]) * BMP_TEMP_SCALE)
+                        if data[12] != BMP_MISSING_TEMP else None),
+            temp_ext_1=((float(data[13]) * BMP_TEMP_SCALE)
+                        if data[13] != BMP_MISSING_TEMP else None),
+            fan_0=float(data[16]) if data[16] != BMP_MISSING_FAN else None,
+            fan_1=float(data[17]) if data[17] != BMP_MISSING_FAN else None,
+        )
+
+
+class BMPInfo(collections.namedtuple(
+    'BMPInfo', "code_block frame_id can_id board_id version buffer_size "
+               "build_date version_string")):
+    """Information returned about a BMP by sver.
+
+    Parameters
+    ----------
+    code_block : int
+        The BMP, on power-up, will execute the first valid block in its flash
+        storage. This value which indicates which 64 KB block was selected.
+    frame_id : int
+        An identifier programmed into the EEPROM of the backplane which
+        uniquely identifies the frame the board is in. Note: This ID is not
+        necessarily the same as a board's frame-coordinate.
+    can_id : int
+        ID of the board in the backplane CAN bus.
+    board_id : int
+        The position of the board in a frame. (This should correspond exactly
+        with a board's board-coordinate.
+    version : float
+        Software version number. (Major version is integral part, minor version
+        is fractional part).
+    buffer_size : int
+        Maximum supported size (in bytes) of the data portion of an SCP packet.
+    build_date : int
+        The time at which the software was compiled as a unix timestamp. May be
+        zero if not set.
+    version_string : string
+        Human readable, textual version information split in to two fields by a
+        "/". In the first field is the kernel (e.g. BC&MP) and the second the
+        hardware platform (e.g. Spin5-BMP).
+    """
+
+
+class ADCInfo(collections.namedtuple(
+    'ADCInfo', "v1_2c v1_2b v1_2a v1_8 v3_3 vpwr "
+               "temp_top temp_btm temp_ext_0 temp_ext_1 fan_0 fan_1")):
+    """ADC data returned by a BMP including voltages and temperature.
+
+    Parameters
+    ----------
+    v1_2a : float
+        Measured voltage on the 1.2 V rail A.
+    v1_2b : float
+        Measured voltage on the 1.2 V rail B.
+    v1_2c : float
+        Measured voltage on the 1.2 V rail C.
+    v1_8 : float
+        Measured voltage on the 1.8 V rail.
+    v3_3 : float
+        Measured voltage on the 3.3 V rail.
+    vpwr : float
```

Comment:
I've been trying to not be too terse with names*: `supply_voltage`?

\* with debatable success.